lib/Test2/Formatter/TAP.pm

```diff
@@ -233,30 +268,41 @@ sub _subtest_event {
     my $self = shift;
     my ($e, $num) = @_;
 
+    # A 'subtest' is a subclass of 'ok'. Let the code that renders 'ok' render
+    # this event.
     my ($ok, @diag) = $self->_ok_event($e, $num);
 
-    return (
-        $ok,
-        @diag
-    ) unless $e->buffered;
+    # If the subtest is not buffered then the sub-events have already been
+    # rendered, we can go ahead and return.
+    return ($ok, @diag) unless $e->buffered;
```

Comment:
Initially that problem didn't exist because each event had a to_tap method to do it... but that was frowned upon even more....